sqlx-core/src/sqlite/connection/executor.rs

```diff
@@ -91,6 +109,12 @@ impl<'c> Executor<'c> for &'c mut SqliteConnection {
             let mut num_arguments = 0;
 
             while let Some((stmt, columns, column_names, last_row_values)) = stmt.prepare(conn)? {
+                // Prepare to reset raw SQLite statement when the handle
+                // is dropped. `StatementResetter` will reliably reset the
+                // statement even if the stream returned from `fetch_many`
+                // is dropped early.
+                let resetter = StatementResetter::new(*stmt);
```

Comment:
you'll want to call it `let _resetter` to avoid unhappy warnings & clippy

Reply:
Fixed and force-pushed